examples/drawing/plot_spring_layout.py

```diff
@@ -0,0 +1,80 @@
+"""
+===================
+Spring Layout
+===================
+
+Draw graphs using the three different spring layout algorithms.
+
+The spring layout is typically generated by the Fruchterman--Reingold force-directed algorithm.
+This algorithm treats edges as springs holding nodes close while treating nodes as repelling objects
+and simulates the system until it reaches equilibrium.
+The algorithm also terminates when it reaches a maximum number of iterations.
+
+NetworkX offers mainly three different kinds of methods based on the same theoretical foundation:
+
+* `nx.spring_layout(G, method="force")`
+
+  * The default for graphs with fewer than 500 nodes in `nx.spring_layout`.
+  * Direct implementation of the Fruchterman--Reingold force-directed algorithm.
+  * Can handle negative edge weights.
+
+* `nx.spring_layout(G, method="energy")`
+
+  * The default for graphs with more than or equal to 500 nodes in `nx.spring_layout`.
+  * It solves an energy-based optimization problem, taking the absolute value of negative edge weights.
+  * Generally produces higher-quality layouts compared to `force`.
+  * Uses gravitational forces acting on each connected component to prevent divergence.
+
+* `nx.nx_agraph.graphviz_layout(G, prog="sfdp")`
+
+  * Uses `sfdp` from GraphViz to compute the layout.
+  * Can also be done with `nx.nx_pydot.graphviz_layout(G, prog="sfdp")`.
```

Comment:
For reasons that are too complicated for the gallery example, I'd vote to remove this line. This is of course true, but there's a subtle difference between pygraphviz and pydot: it is possible to install pydot without Graphviz being installed *but* using the layouts *would* require Graphviz. pygraphviz on the other hand can only be successfully installed if Graphviz is already available on the system, so a working `pygraphviz` installation == a working interface to the sfdp layout.

All that's to say, I'd vote to keep the emphasis on `nx_agraph` and silently downplay `nx_pydot` here. Not a blocker though!